src/plugins/intel_gpu/src/plugin/ops/batch_to_space.cpp

```diff
@@ -23,23 +23,10 @@ static void CreateBatchToSpaceOp(ProgramBuilder& p, const std::shared_ptr<ov::op
     std::vector<cldnn::tensor> tensor_inputs;
     tensor_inputs.reserve(3);
 
-    bool non_constant_input = false;
-    for (size_t i = 1; i < 4; ++i) {
-        auto inConst = ov::as_type_ptr<ov::op::v0::Constant>(op->get_input_node_shared_ptr(i));
-
-        bool is_const_input = (inConst != nullptr);
-        OPENVINO_ASSERT((i == 1) || (i >= 2 && non_constant_input != is_const_input),
-            "[GPU] Unsupported mixed node with constant and parameter in ", op->get_friendly_name(), " (", op->get_type_name(), ")");
-
-        if (!inConst) {
-            non_constant_input = true;
-        }
-    }
-
     auto output_pshape = op->get_output_partial_shape(0);
     auto out_size = output_pshape.is_static() ? tensor_from_dims(output_pshape.to_shape()) : cldnn::tensor();
 
-    if (non_constant_input) {
```

Comment:
Could you explain the behavior? I cannot understand how it would work. For example, if the model is static but input is not constant (`!op->is_dynamic() && non_constant_input`), I guess it should go through the "then" block of `if`. (`batch_to_space(layerName, inputs, out_size)`) But with your change, it would go to "else" block of `if`. I guess that wouldn't work.

Reply:
This checking code whether constant or not is old (about 1.5 years old) and not necessary any more because we don't care whether input is constant node or parameter node at recent OpenVINO.
Instead of checking constant node, checking dynamic is necessary because dummy is required for dynamic case.

Reply:
There is a code `std::vector<int32_t> sizes = inConst->cast_vector<int32_t>();` below. It assumes that the input is a constant node. It conflicts with what you are saying ("in recent OpenVINO, we don't care if the input is a constant node or a parameter node"). If it is not needed to check constantness, isn't it necessary to remove the whole "else" block?

Reply:
Sorry for my misunderstanding. The spaceToBatch and batchToSpace ops need to check constantness to support legacy for old_shape_infer. PR is updated.